WooCommerce/Classes/ViewRelated/Products/Add Product/FirstProductCreated/FirstProductCreatedViewModel.swift

```diff
@@ -0,0 +1,44 @@
+import Yosemite
+
+final class FirstProductCreatedViewModel: ObservableObject {
+    @Published var isSharePopoverPresented = false
+
+    let productURL: URL
+    let productName: String
+    let showShareProductButton: Bool
+    let shareSheet: ShareSheet
+
+    var launchAISharingFlow: (() -> Void)?
```

Comment:
Nit: As commented above, we can make this closure private if we inject it in the view model's initializer.

Also - the many different code paths make it hard to follow the logic. Please consider one of these options:
- Rename the closure to something more generic like `startShareCoordinator` and add a comment to explain why it's needed. The reason that we are not using it just for the AI flow but also for the native share sheet on iPhones.
- Keep this closure for just the AI flow, and add an extra property to show the share sheet for iPhones directly from the `FirstProductCreatedView`. This would ensure readability for the logic of the share flows.

Reply:
Good idea. 💡 

I followed the second option in https://github.com/woocommerce/woocommerce-ios/pull/9901/commits/db01361c4f06265f3fdb530a4c9a7a6c45e4e548 and updated tests in https://github.com/woocommerce/woocommerce-ios/pull/9901/commits/816a6ae06ffd519d254b7cd60b762039781cd1f2